app/controllers/user/mygardens/plants_controller.rb

```diff
@@ -8,12 +8,16 @@ def create
 		plant = Plant.find(params[:plant_id])
     garden.plants << plant
     garden.save
-    add_notification(garden, plant)
+    Thread.new do
+      add_notification(garden, plant)
+    end
+    flash[:notice] = "We're updating your calendar with the harvest time of your #{plant.name.downcase}"
 		redirect_to("/user/mygardens/#{garden.id}")
   end
 
   def destroy
     garden_plant = GardenPlant.where("garden_id= #{params[:garden_id]} AND plant_id= #{params[:plant_id]}").first
+    remove_notifications(garden_plant, current_user)
```

Comment:
deletes all `events` in the `calendar` associated with the `plant` being removed from the `garden`